### ISSUE TITLE
[REGRESSION] `db-service v1.12.1` broke deep inserts

### ISSUE THREAD

**Issue by hakimio** [cap-js/cds-dbs#811]
### Description of erroneous behavior 

Following works fine with `1.12.0` but breaks with `1.12.1`:
```js
const result = await INSERT.into('MyService.MyEntity').entries({
  foo: 'bar',
  nestedEntity: {
    bar: 'baz'
  }
});
```

With `1.12.0` proper result is returned, with `1.12.1` result is `0` and nothing is inserted.

### Details about your project

| Name | Version |
|:------------------|---------------------------------------|
| OData version     | v4                               |
| Node.js version   | v20.13.1                               |
| @sap/cds          | 7.9.4                                 |
| @sap/cds-compiler | 4.9.8                                 |
| @sap/cds-dk       | 7.9.7                                 |
| @cap-js/postgres | 1.10 |

**Comment by hakimio** [comment]
@David-Kunz

**Comment by David-Kunz** [comment]
Hi @hakimio ,

Can you kindly show us your model? The following example works as expected:

```cds
namespace myNamespace;
using { cuid } from '@sap/cds/common';

entity MyEntity : cuid {
  name: String(100);
  description: String(500);
  nestedEntities: Composition of many NestedEntity on nestedEntities.parent = $self;
}

entity NestedEntity : cuid {
  name: String(100);
  value: Integer;
  parent: Association to MyEntity;
}
```

```js
module.exports = srv => {

  srv.on('READ', 'MyEntity', async (req, next) => {

    const res = await INSERT.into('myNamespace.MyEntity').entries({
      ID: 'a11fb6f1-36ab-46ec-b00c-d379031e817a',
      nestedEntities: [
        { ID: 'b22fb6f1-36ab-46ec-b00c-d379031e817a', value: 5 }
      ]
    })

    console.log(res) // InsertResult { results: [ { changes: 1, lastInsertRowid: 1 } ] }
    
    return next()
  })

}
```

Thanks and best regards,
David

**Comment by hakimio** [comment]
The `nestedEntity` does **NOT** require an ID when the parent entity is created.

```cds
entity MyEntities: cuid {
    foo: String;
    nestedEntity: Composition of NestedEntities;
}

entity NestedEntities: cuid {
    bar: String;
}
```

This does not work anymore (**no ID for nested are parent entity**):
```ts
const result = await INSERT.into('MyService.MyEntity').entries({
  foo: 'bar',
  nestedEntity: {
    bar: 'baz'
  }
});
```

**Comment by David-Kunz** [comment]
Hi @hakimio ,

Your example works too:

```cds
using { cuid } from '@sap/cds/common';

entity MyEntities: cuid {
    foo: String;
    nestedEntity: Composition of NestedEntities;
}

entity NestedEntities: cuid {
    bar: String;
}
```

```js
module.exports = srv => {

  srv.before('READ', 'SMyEntities', async req => {
    const x = await INSERT.into('MyEntities').entries({
      nestedEntity: {
        bar: 'bar'
      }
    })
    const y = await INSERT.into('MyEntities').entries({})
    console.log(await SELECT.from('MyEntities', x[0])) // find the entry
    console.log(await SELECT.from('MyEntities', y[0])) // finds the entry
  })

}
```

Can you kindly adjust the example to reproduce your issue?

Thanks and best regards,
David

**Comment by hakimio** [comment]
@David-Kunz what version of `@sap/cds` are you using? Have you tested with `v7`?
What db implementation are you using (type and version)?

**Comment by David-Kunz** [comment]
Hi @hakimio ,

I can reproduce it with @sap/cds 7. Will have a look.

**Comment by David-Kunz** [comment]
Sorry, @hakimio , it works, also with version 7. Can you reproduce it also with SQLite or does it only occur with Postgres?

**Comment by hakimio** [comment]
I only tested with Postgres. 
I'll try to debug and see if I can find out what exactly is going on sometime this week.

**Comment by David-Kunz** [comment]
Also can't reproduce it with Postgres.
```
cds version

@cap-js/cds-types: 0.2.0
@cap-js/db-service: 1.12.1
@cap-js/hana: 1.2.0
@cap-js/postgres: 1.10.0
@cap-js/sqlite: 1.7.3
@sap/cds: 7.9.5
@sap/cds-compiler: 4.9.8
@sap/cds-fiori: 1.2.7
@sap/cds-foss: 5.0.1
@sap/cds-mtxs: 2.0.0
Node.js: v22.9.0
```

**Comment by David-Kunz** [comment]
However, `x === 0`, even though it's inserted.

**Comment by hakimio** [comment]
That's still a bug.

**Comment by David-Kunz** [comment]
Yes, it was fixed with https://github.com/cap-js/cds-dbs/pull/803

**Comment by David-Kunz** [comment]
Let me check when we can release this.

**Comment by David-Kunz** [comment]
Will soon be released.

**Comment by David-Kunz** [comment]
And just to be sure: Is your data really not inserted? Can you double check with `DEBUG=sql`?

**Comment by hakimio** [comment]
Most likely what was happening was when the result was `0` instead of expected object, my app would throw an error and the transaction for the request would be reverted.

**Comment by David-Kunz** [comment]
I see, this will be fixed in the next version.

**Comment by hakimio** [comment]
Fixed in [1.13.0](https://github.com/cap-js/cds-dbs/releases/tag/db-service-v1.13.0).
